src/cascadia/TerminalApp/CascadiaSettings.cpp

```diff
@@ -532,33 +524,37 @@ GUID CascadiaSettings::_GetProfileForArgs(const NewTerminalArgs& newTerminalArgs
         {
             bool wasGuid = false;
 
-            try
+            // Do a quick heuristic check - is the profile 38 chars long (the
+            // length of a GUID string), and does it start with '{'? Because if
+            // it doesn't, it's _definitely_ not a GUID.
+            if (newTerminalArgs.Profile().size() == 38 && newTerminalArgs.Profile()[0] == L'{')

```

Comment:
there may be value in caching `newTerminalArgs.Profile()` upfront in a local just to avoid the repeated calls through the projection and perhaps even QIs.